libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -1028,25 +1117,198 @@ void test_MQTT_Restore_Session_Complete_Incoming_Publish( void )
     TEST_ASSERT_NOT_EQUAL( -1, networkContext.socketDescriptor );
     TEST_ASSERT_NOT_NULL( networkContext.pSsl );
 
-    /* Re-establish the persistent session with the broker by connecting with "clean session" flag set to 0. */
+    /* Re-establish a connection with the broker to resend the PUBLISH packet. */
     establishMqttSession( &context, &networkContext, false, &persistentSession );
 
-    /* Verify that the session was resumed. */
-    TEST_ASSERT_TRUE( persistentSession );
+    /* Obtain the packet ID of the PUBLISH packet that didn't complete in the previous connection. */
+    MQTTStateCursor_t cursor = 0u;
```

Comment:
Nit: it might be cleaner to use `MQTT_STATE_CURSOR_INITIALIZER` since that's what it's for
```suggestion
    MQTTStateCursor_t cursor = MQTT_STATE_CURSOR_INITIALIZER;
```